### ISSUE TITLE
Nested notation applicable on intermediate elements not found

### ISSUE THREAD

**Issue by JeroenBos** [JeroenBos/ASDE#5]
Context: We're matching a notation, and we select some leaves for an element with nested notation.

Normally, an AST is computed for those selected leaves. This takes `N` round, say, and results in a single node, the default result node. However, if that last parsing round didn't use the nested notation, the element won't accept that AST. Thus, we try to force parsing using the nested notation. 

The problem is that we only do that on the single default result node, and if that fails, the notation is also forced on the selected leaves, but if that fails we don't try on any of the intermediate states after `n` round of normal parsing, where `0 < n < N`, on which the nested notation may be applicable.

Basically, I just implemented those `n == 0` and `n == N` cases (where `n == N` is preferred), but maybe I want to attempt it on the intermediate nodes as well.

There is no need to solve this problem (for now), I just wanted to mention it. A solution would require another overhaul, but I might state this to be by design. To be determined
